BinaryTree.java

```diff
@@ -0,0 +1,165 @@
+        public int getValue() {
+            return value;
+        }
+
+        public TreeNode getLeft() {
+            return left;
+        }
+
+        @Override
+        public String toString() {
+            return "TreeNode{" +
+                    "value=" + value +
+                    '}';
+        }
+
+        public TreeNode getRight() {
+            return right;
+        }
+
+    }
+
+    /* Function to print all ancestors of purposed value
+    * @param node - root node
+    * @param value - value of the node on which we want to print ancestors
+    * @return true, if value exists in this tree, otherwise - false*/
+    boolean printAllAncestors(TreeNode node, int value)
+    {
+         //if tree is empty returns false
+        if (node == null)
+            return false;
+
+        //if value exist in a tree
+        if (node.getValue() == value)
+            return true;
+
+        //if value is present in left/right subtree
+        // then we'll print it
+        if (printAllAncestors(node.left, value) || printAllAncestors(node.right, value))
+        {
+            System.out.print(node.value + " ");
+            return true;
+        }
+
+        //else return false
+        return false;
+    }
+
+    /*Traverse tree in preorder
+    * @param root - tree root*/
+    public void preorderTraversal (TreeNode root){ //метод для обхода в ширину
+        if (root != null){
+            System.out.print(root.getValue() + " ");
+            preorderTraversal (root.left);
+            preorderTraversal (root.right);
+        }
+    }
+
+    /*Populates tree with the array in order
+    * @param array - array which will fill the tree*/
+    public boolean populateTree(int[] array) {
+        for (int i = 0; i < array.length; i++) {
+            this.insert(array[i]);
+        }
+
+        return true;
+    }
+
+    /*@param value - value to insert*/
+    public void insert(int value) {
+        root = insert(root, value);
+    }
+
+    /*Inserts value to the tree
+    * @param node - root
+    * @param value - value to insert
+    * @return root*/
+    private static TreeNode insert(TreeNode node, int value) {
+        if (node == null) {
+            return new TreeNode(value);
+        }
+        if (value < node.value) {
+            node.left = insert(node.left, value);
+        } else if (value > node.value) {
+            node.right = insert(node.right, value);
+        }
+        return node;
+    }
+
+    /*Finds lowest common ancestor of two nodes
+    * @param root
+    * @param a - first node value
+    * @param b - second node value
+    * @return lowest common ancestor if both nodes are present
+    * in the tree, if one is not present - returns existing node, as LCA
+    * if both don't exist - returns null
+    * if a key is ancestor of other, then the ancestor key becomes LCA*/
+    public TreeNode findLowestCommonAncestor(TreeNode root, int a, int b) {
+        // Base case
+        if (root == null) return null;

```

Comment:
Please try to avoid both single-line if statements and ones without {}. Always including brackets does not increase code size that much, but highly reduces mental power needed to read code, and thus makes it easier to avoid silly mistakes while modifying it in the future.